gax/pagination.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google Inc.  All rights reserved
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GAPIC_GENERATOR_CPP_GAX_PAGINATION_H_
+#define GAPIC_GENERATOR_CPP_GAX_PAGINATION_H_
+
+#include <google/protobuf/repeated_field.h>
+#include <functional>
+#include <iterator>
+#include <string>
+
+namespace google {
+namespace gax {
+
+template <typename ElementType, typename PageType>
+class PageResult {
+  using ElementAccessor =
+      std::function<::google::protobuf::RepeatedPtrField<ElementType>*(
```

Comment:
Should `::google::protobuf::RepeatedPtrField<ElementType>` also become an alias (it is used 3 times in this class)?

Reply:
Removed and deduced by template magic.